pkg/restmapper/dynamicrestmapper.go

```diff
@@ -15,30 +15,64 @@
 package restmapper
 
 import (
+	"sync"
+	"time"
+
 	"k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/restmapper"
 )
 
+// TODO(estroz): do we want to return a wait.ErrWaitTimeout if backoff duration
+// reaches a maximum?
```

Comment:
I would say yes. Backoff duration hitting its maximum is, in the end, a wait timeout.